rclcpp/src/rclcpp/node_interfaces/node_parameters.cpp

```diff
@@ -253,12 +523,16 @@ NodeParameters::get_parameters(const std::vector<std::string> & names) const
   std::vector<rclcpp::Parameter> results;
```

Comment:
@wjwwood nit: consider doing `results.reserve(names.size())` to avoid incremental reallocations below.